program/scripts/PDBminer_functions.py

```diff
@@ -215,70 +133,66 @@ def get_structure_df(uniprot_id):
             
             return structure_df
     
-    #if there are PDBs these should be captured. 
-    else: 
-        #empty list for collection of PDBs 
-        currated_pdb_list = []
-        #create an empty df
+    else:
+        structures = response_text['structures']
+        
         structure_df = pd.DataFrame()
-    
-        #make a for loop to populate the empty df with metadata for each pdb
-        for pdb in pdb_list:
-            #retrive all metadata        
-            structure_metadata = get_structure_metadata(pdb)
-            if type(structure_metadata) == tuple:
-                
-                data = [(structure_metadata[0]), (structure_metadata[1]), (structure_metadata[2])]
-            
-                structure_metadata_df = pd.DataFrame([data], columns = ['deposition_date', 
-                                                                      'experimental_method', 
-                                                                      'resolution'])
-                
-                structure_df = pd.concat([structure_df, structure_metadata_df])               
-                currated_pdb_list.append(pdb)
         
-            else: 
-                continue
-                
-        #Add pdb ID & uniprot ID as columns
-        structure_df.index = currated_pdb_list
-        structure_df.insert(0, "uniprot_id", [uniprot_id]*len(structure_df), True)
+        pdb = []
+        deposition_date = []
+        experimental_method = [] 
+        resolution = []
         
-        #Split based on experimental method
-        method_dfs = [x for _, x in structure_df.groupby(structure_df['experimental_method'])]
+        for structure in structures:
+            pdb.append(structure['summary']['model_identifier'].upper())
+            deposition_date.append(structure['summary']['created'])
+            experimental_method.append(structure['summary']['experimental_method'])
+            resolution.append(structure['summary']['resolution'])
+            
+    #structure_df.index = pdb
+    structure_df.insert(0, "pdb", pdb, True)
+    structure_df.insert(1, "uniprot_id", [uniprot_id]*len(structure_df), True)
+    structure_df.insert(2, "deposition_date", deposition_date, True)
+    structure_df.insert(3, "experimental_method", experimental_method, True)
+    structure_df.insert(4, "resolution", resolution, True)
+    
+    structure_df = structure_df.set_index('pdb')
+    
+    #Split based on experimental method
+    method_dfs = [x for _, x in structure_df.groupby(structure_df['experimental_method'])]
+    
+    #Sort within the method
```

Comment:
I see that this could be a smarter was to sort the rows. 

There are many different experimental methods, and I am not finding an easy way to say that "NMR" should be part of the word rather than the whole thing. 

A solution: 
```
for method_df in method_dfs:
        method_df.sort_values(["resolution", "deposition_date"], ascending=[True, False], inplace=True)
        
        rank_offset = {
            'X-RAY DIFFRACTION': 1000,
            'ELECTRON': 2000,
            'NMR': 3000}.get(method_df.experimental_method.iloc[0], 4000)
        
        ranks = range(rank_offset, rank_offset + len(method_df))
        method_df.insert(0, "Rank", sorted(ranks), True)
 
structure_df = pd.concat(method_dfs)
```

However, this assigns rank 4000 + to NMR and CRYO structures.

Reply:
<img width="258" alt="image" src="https://github.com/ELELAB/PDBminer/assets/84312237/43aadbe6-1bd7-45a2-97c3-58d1fd14a23e">

Reply:
```
method_dfs = [x for _, x in structure_df.groupby(structure_df['experimental_method'])]
    
    for method_df in method_dfs:
        method_df.sort_values(["resolution", "deposition_date"], ascending=[True, False], inplace=True)
        
        rank_offset = {
            'X-RAY DIFFRACTION': 1000,
            'ELECTRON MICROSCOPY': 2000,
            'ELECTRON CRYSTALLOGRAPHY': 3000,
            'SOLUTION NMR': 4000,
            'SOLID-STATE NMR': 5000}.get(method_df.experimental_method.iloc[0], 6000)
        
        ranks = range(rank_offset, rank_offset + len(method_df))
        method_df.insert(0, "Rank", sorted(ranks), True)
    
    structure_df = pd.concat(method_dfs)
```